usb.c

```diff
@@ -246,50 +246,56 @@ static int usb_open(const struct iio_device *dev,
 	const struct iio_context *ctx = iio_device_get_context(dev);
 	struct iio_context_pdata *ctx_pdata = iio_context_get_pdata(ctx);
 	struct iio_device_pdata *pdata = iio_device_get_pdata(dev);
+	struct iiod_client *client = ctx_pdata->iiod_client;
+	unsigned int timeout;
 	int ret = -EBUSY;
 
 	iio_mutex_lock(ctx_pdata->ep_lock);
 
 	pdata->io_ctx.cancelled = false;
 
 	if (pdata->opened)
-		goto out_unlock;
+		goto err_unlock;
 
 	ret = usb_reserve_ep_unlocked(dev);
 	if (ret)
-		goto out_unlock;
+		goto err_unlock;
 
 	ret = usb_open_pipe(ctx_pdata, pdata->io_ctx.ep->pipe_id);
 	if (ret) {
 		dev_perror(dev, -ret, "Failed to open pipe");
-		usb_free_ep_unlocked(dev);
-		goto out_unlock;
+		goto err_free_ep;
 	}
 
 	iio_mutex_lock(pdata->lock);
 
-	ret = iiod_client_open_unlocked(ctx_pdata->iiod_client, &pdata->io_ctx,
-			dev, samples_count, cyclic);
-
-	if (!ret) {
-		unsigned int remote_timeout =
-			usb_calculate_remote_timeout(ctx_pdata->timeout_ms);
+	ret = iiod_client_open_unlocked(client, &pdata->io_ctx,
+					dev, samples_count, cyclic);
+	if (ret)
+		goto err_close_pipe;
```

Comment:
`iio_mutex_unlock(pdata->lock);` ? Also in the error handling of `iiod_client_set_timeout()`...

Reply:
Good catch.